api/modelmanager/modelmanager.go

```diff
@@ -104,6 +104,9 @@ func convertParamsModelInfo(modelInfo params.ModelInfo) (base.ModelInfo, error)
 		Owner:           ownerTag.Id(),
 		Life:            string(modelInfo.Life),
 	}
+	if modelInfo.AgentVersion != nil {
```

Comment:
You don't need this check since the result of not checking is exactly the same as checking.

Just put the AgentVersion part into the struct initializer above.